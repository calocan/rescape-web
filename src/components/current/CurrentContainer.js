import {connect} from 'react-redux';
import Current from './Current';
import * as R from 'ramda';
import {createSelector} from 'reselect';
import {
  makeActiveUserSelectedRegionAndSettingsSelector,
} from 'selectors/storeSelectors';
import {makeBrowserProportionalDimensionsSelector, makeMergeDefaultStyleWithProps} from 'selectors/styleSelectors';
import {mergeDeep} from 'rescape-ramda';
import {loadingCompleteStatus, makeTestPropsFunction} from 'rescape-helpers-component';
import {bindActionCreators} from 'redux';

/**
 * Combined selector that:
 * Selects the active user and user's selected region.
 * Merges the component dimensions to the browser with the default styles.
 * @param {Object} state The redux state
 * @param {Object} [props] The optional props to override the state.
 * @returns {Object} The state and own props mapped to props for the component
 */
export const mapStateToProps = (state, props) => {
  const {style, ...data} = props
  return createSelector(
    [
      makeActiveUserSelectedRegionAndSettingsSelector(),
      makeMergeDefaultStyleWithProps(),
      makeBrowserProportionalDimensionsSelector()
    ],
    (stateData, defaultStyle, browserProportionalStyle) => ({
      // No current graphql queries, pass the winnowed state
      // It might turn out that Current doesn't need anything because it simply renders child containers
      // Merge the browser dimensions with the props
      // props from the parent contain style instructions
      // TODO we need to set width and height proportional to the browser dimensions, not equal to
      data: R.mergeAll([stateData, loadingCompleteStatus, data]),
      style: R.mergeAll([defaultStyle, browserProportionalStyle, style])
    })
  )(state, props);
}

export const mapDispatchToProps = (dispatch, ownProps) => {
  return bindActionCreators({}, dispatch);
};



export default connect(mapStateToProps, mapDispatchToProps)(Current);
