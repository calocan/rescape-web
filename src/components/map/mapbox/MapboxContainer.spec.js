import {
  asyncPropsFromSampleStateAndContainer,
  propsFromSampleStateAndContainer, wrapWithMockGraphqlAndStore
} from 'helpers/testHelpers';
import {testPropsMaker} from 'components/map/mapbox/MapboxContainer';
import {testPropsMaker as currentPropsMaker} from 'components/current/CurrentContainer';
import {testPropsMaker as regionPropsMaker} from 'components/region/RegionContainer';
import {eMap} from 'helpers/componentHelpers';
import MapboxContainer from 'components/map/mapbox/MapboxContainer';
import * as R from 'ramda';
import {c as cCurrent} from 'components/current/Current';
import Region, {c as cRegion} from 'components/region/Region';

describe('MapboxContainer', () => {
  test('render', async () => {
    // Build up the correct parent props from Current and Region
    const currentProps = propsFromSampleStateAndContainer(currentPropsMaker, {});
    const regionProps = await asyncPropsFromSampleStateAndContainer(regionPropsMaker, currentProps.views[cCurrent.currentRegion]);
    const regionViews = Region.views(regionProps).views

    const parentProps = R.merge({
      style: {
        width: 500,
        height: 500
      }
    }, regionViews[cRegion.regionMapbox]);

    // Get the test props for MapboxContainer
    const props = propsFromSampleStateAndContainer(testPropsMaker, parentProps);

    expect(props).toMatchSnapshot();
    const [mapboxContainer] = eMap([MapboxContainer]);
    const wrapper = wrapWithMockGraphqlAndStore(mapboxContainer(parentProps));
    const current = wrapper.find('Mapbox');
    expect(R.keys(current.props())).toEqual(['data', 'views']);
  });
});
