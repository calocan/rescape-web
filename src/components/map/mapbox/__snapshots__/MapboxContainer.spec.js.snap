// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapboxContainer mapStateToProps 1`] = `
Object {
  "data": Object {
    "error": false,
    "loading": false,
    "mapbox": Object {
      "iconAtlas": "data/location-icon-atlas.png",
      "mapboxApiAccessToken": "pk.eyJ1IjoiY2Fsb2NhbiIsImEiOiJjaXl1aXkxZjkwMG15MndxbmkxMHczNG50In0.07Zu3XXYijL6GJMuxFtvQg",
      "preventStyleDiffing": false,
      "showCluster": "true",
      "showZoomControls": true,
    },
    "region": Object {
      "id": "belgium",
      "mapbox": Object {
        "viewport": Object {
          "latitude": 50.5915,
          "longitude": 2.0165,
          "zoom": 5,
        },
      },
      "name": "Belgium",
    },
    "settings": Object {
      "api": Object {
        "host": "localhost",
        "port": "3000",
        "protocol": "http",
        "root": "/graphql/",
      },
      "cycle": Object {
        "drivers": Object {
          "api": "HTTP",
        },
      },
      "domain": "localhost",
      "graphcool": Object {
        "authTokenKey": "graphcool-auth-token",
        "linksPerPage": 5,
        "serviceIdKey": "cjajyycub38710185wt87zsm8",
        "userId": "graphcool-user-id",
      },
      "mapbox": Object {
        "iconAtlas": "data/location-icon-atlas.png",
        "mapboxApiAccessToken": "pk.eyJ1IjoiY2Fsb2NhbiIsImEiOiJjaXl1aXkxZjkwMG15MndxbmkxMHczNG50In0.07Zu3XXYijL6GJMuxFtvQg",
        "preventStyleDiffing": false,
        "showCluster": "true",
        "showZoomControls": true,
      },
      "markers": Object {},
      "overpass": Object {
        "cellSize": 100,
        "sleepBetweenCalls": 1000,
      },
    },
    "store": Object {
      "region": Object {
        "geojson": Object {
          "osm": Object {
            "features": Array [
              Object {
                "geometry": Object {
                  "coordinates": Array [
                    Array [
                      2.3372032,
                      48.8404254,
                    ],
                    Array [
                      2.3372715,
                      48.8406677,
                    ],
                    Array [
                      2.3373962,
                      48.8410543,
                    ],
                    Array [
                      2.3380024,
                      48.8422009,
                    ],
                    Array [
                      2.3385107,
                      48.8431117,
                    ],
                    Array [
                      2.3389787,
                      48.8439785,
                    ],
                    Array [
                      2.3392577,
                      48.8444515,
                    ],
                    Array [
                      2.3402917,
                      48.8463852,
                    ],
                    Array [
                      2.3405962,
                      48.8469464,
                    ],
                    Array [
                      2.3407859,
                      48.8472594,
                    ],
                    Array [
                      2.3411775,
                      48.8479028,
                    ],
                    Array [
                      2.3422987,
                      48.8497393,
                    ],
                    Array [
                      2.34276,
                      48.8507232,
                    ],
                    Array [
                      2.3429223,
                      48.8510339,
                    ],
                    Array [
                      2.3430309,
                      48.8511954,
                    ],
                    Array [
                      2.3432019,
                      48.8514041,
                    ],
                    Array [
                      2.3434513,
                      48.851675,
                    ],
                    Array [
                      2.3436619,
                      48.851859,
                    ],
                    Array [
                      2.3439871,
                      48.8520933,
                    ],
                    Array [
                      2.3443787,
                      48.8523492,
                    ],
                    Array [
                      2.3447636,
                      48.8525821,
                    ],
                    Array [
                      2.3454415,
                      48.8529788,
                    ],
                    Array [
                      2.3458338,
                      48.8532131,
                    ],
                    Array [
                      2.346153,
                      48.8534072,
                    ],
                    Array [
                      2.3465238,
                      48.8536327,
                    ],
                    Array [
                      2.3468805,
                      48.8538939,
                    ],
                    Array [
                      2.3472815,
                      48.8542327,
                    ],
                    Array [
                      2.3476597,
                      48.8546377,
                    ],
                  ],
                  "type": "LineString",
                },
                "id": "way/29735335",
                "properties": Object {
                  "@changeset": "40980697",
                  "@id": "way/29735335",
                  "@timestamp": "2016-07-23T22:33:32Z",
                  "@uid": "158826",
                  "@user": "cquest",
                  "@version": "34",
                  "electrified": "contact_line",
                  "frequency": "0",
                  "gauge": "1435",
                  "importance": "regional",
                  "layer": "-5",
                  "name": "RER B",
                  "operator": "RATP",
                  "railway": "rail",
                  "source": "STIF",
                  "tunnel": "yes",
                  "usage": "main",
                  "voltage": "1500",
                },
                "type": "Feature",
              },
              Object {
                "geometry": Object {
                  "coordinates": Array [
                    Array [
                      2.3580437,
                      48.8489285,
                    ],
                    Array [
                      2.3569883,
                      48.8493417,
                    ],
                    Array [
                      2.3566417,
                      48.8494774,
                    ],
                    Array [
                      2.3498091,
                      48.8517476,
                    ],
                    Array [
                      2.3481873,
                      48.85241,
                    ],
                    Array [
                      2.3468858,
                      48.8530098,
                    ],
                    Array [
                      2.3465228,
                      48.8531476,
                    ],
                    Array [
                      2.3454232,
                      48.8534731,
                    ],
                  ],
                  "type": "LineString",
                },
                "id": "way/334870919",
                "properties": Object {
                  "@changeset": "34996941",
                  "@id": "way/334870919",
                  "@timestamp": "2015-10-31T17:43:46Z",
                  "@uid": "53964",
                  "@user": "_Mathieu_",
                  "@version": "4",
                  "electrified": "contact_line",
                  "frequency": "0",
                  "gauge": "1435",
                  "importance": "regional",
                  "layer": "-1",
                  "maxspeed": "60",
                  "name": "RER C - Ligne de Musée d'Orsay à Austerlitz",
                  "railway": "rail",
                  "tracks": "2",
                  "tunnel": "yes",
                  "usage": "main",
                  "voltage": "1500",
                },
                "type": "Feature",
              },
              Object {
                "geometry": Object {
                  "coordinates": Array [
                    Array [
                      2.3454232,
                      48.8534731,
                    ],
                    Array [
                      2.344675,
                      48.8536946,
                    ],
                    Array [
                      2.3440745,
                      48.8539094,
                    ],
                    Array [
                      2.343693,
                      48.8540843,
                    ],
                    Array [
                      2.3431388,
                      48.854372,
                    ],
                    Array [
                      2.342491,
                      48.8547437,
                    ],
                    Array [
                      2.3402755,
                      48.8563525,
                    ],
                    Array [
                      2.3379956,
                      48.8575207,
                    ],
                    Array [
                      2.3329547,
                      48.8584095,
                    ],
                    Array [
                      2.3275778,
                      48.8601289,
                    ],
                  ],
                  "type": "LineString",
                },
                "id": "way/377746412",
                "properties": Object {
                  "@changeset": "34996941",
                  "@id": "way/377746412",
                  "@timestamp": "2015-10-31T17:43:41Z",
                  "@uid": "53964",
                  "@user": "_Mathieu_",
                  "@version": "1",
                  "electrified": "contact_line",
                  "frequency": "0",
                  "gauge": "1435",
                  "importance": "regional",
                  "layer": "-1",
                  "maxspeed": "50",
                  "name": "RER C - Ligne de Musée d'Orsay à Austerlitz",
                  "railway": "rail",
                  "tracks": "2",
                  "tunnel": "yes",
                  "usage": "main",
                  "voltage": "1500",
                },
                "type": "Feature",
              },
              Object {
                "geometry": Object {
                  "coordinates": Array [
                    2.3454363,
                    48.8534806,
                  ],
                  "type": "Point",
                },
                "id": "node/3572156993",
                "properties": Object {
                  "@changeset": "42352341",
                  "@id": "node/3572156993",
                  "@timestamp": "2016-09-22T16:51:49Z",
                  "@uid": "1429581",
                  "@user": "Guillaume Mortier",
                  "@version": "3",
                  "STIF:zone": "1",
                  "name": "Saint-Michel Notre-Dame",
                  "official_name": "ST MICHEL NOTRE DAME",
                  "operator": "SNCF;RATP",
                  "railway": "station",
                  "ref": "RER B, C",
                  "ref:FR:RATP": "5604",
                  "ref:FR:uic8": "87547315;87785436",
                  "ref:SNCF": "RER",
                  "ref:SNCF:RER": "C",
                  "type:RATP": "rer",
                  "uic_ref": "8754731;8778543",
                  "wheelchair": "yes",
                  "wikipedia": "fr:Gare Saint-Michel - Notre-Dame",
                },
                "type": "Feature",
              },
              Object {
                "geometry": Object {
                  "coordinates": Array [
                    2.3402917,
                    48.8463852,
                  ],
                  "type": "Point",
                },
                "id": "node/327432589",
                "properties": Object {
                  "@changeset": "40980697",
                  "@id": "node/327432589",
                  "@timestamp": "2016-07-23T22:33:35Z",
                  "@uid": "158826",
                  "@user": "cquest",
                  "@version": "12",
                  "STIF:zone": "1",
                  "name": "Luxembourg",
                  "operator": "RATP",
                  "railway": "station",
                  "ref": "RER B",
                  "ref:FR:RATP": "1811",
                  "type:RATP": "rer",
                  "uic_ref": "8775861",
                  "wheelchair": "yes",
                  "wikipedia": "fr:Gare du Luxembourg",
                },
                "type": "Feature",
              },
              Object {
                "geometry": Object {
                  "coordinates": Array [
                    2.346153,
                    48.8534072,
                  ],
                  "type": "Point",
                },
                "id": "node/270191809",
                "properties": Object {
                  "@changeset": "42352341",
                  "@id": "node/270191809",
                  "@timestamp": "2016-09-22T16:51:50Z",
                  "@uid": "1429581",
                  "@user": "Guillaume Mortier",
                  "@version": "29",
                  "STIF:zone": "1",
                  "name": "Saint-Michel Notre-Dame",
                  "official_name": "ST MICHEL NOTRE DAME",
                  "operator": "SNCF;RATP",
                  "public_transport": "stop_position",
                  "railway": "stop",
                  "ref": "RER B, C",
                  "ref:FR:RATP": "5604",
                  "ref:FR:uic8": "87547315;87785436",
                  "ref:SNCF": "RER",
                  "ref:SNCF:RER": "C",
                  "type:RATP": "rer",
                  "uic_ref": "8754731;8778543",
                  "wheelchair": "yes",
                  "wikipedia": "fr:Gare Saint-Michel - Notre-Dame",
                },
                "type": "Feature",
              },
              Object {
                "geometry": Object {
                  "coordinates": Array [
                    2.3569883,
                    48.8493417,
                  ],
                  "type": "Point",
                },
                "id": "node/3809893083",
                "properties": Object {
                  "@changeset": "34971642",
                  "@id": "node/3809893083",
                  "@timestamp": "2015-10-30T12:46:44Z",
                  "@uid": "53964",
                  "@user": "_Mathieu_",
                  "@version": "1",
                  "highway": "crossing",
                },
                "type": "Feature",
              },
            ],
          },
        },
        "id": "belgium",
      },
    },
    "user": Object {
      "id": "belgiumManager",
      "isActive": true,
      "name": "Belgium Manager",
      "permissions": Array [
        "manage",
      ],
      "regions": Array [
        Object {
          "id": "belgium",
          "isSelected": true,
        },
      ],
    },
    "viewport": Object {
      "bearing": 0,
      "isDragging": false,
      "latitude": 50.5915,
      "longitude": 2.0165,
      "pitch": 40,
      "startDragLngLat": null,
      "zoom": 5,
    },
  },
  "style": Object {
    "MozAppearance": "none",
    "WebkitAppearance": "none",
    "color": "black",
    "fontFamily": "inherit",
    "fontSize": "inherit",
    "height": -32,
    "textDecoration": "none",
    "width": -32,
  },
}
`;

exports[`MapboxContainer query 1`] = `
Object {
  "data": Object {
    "store": Object {
      "__typename": "Store",
      "region": Object {
        "__typename": "Region",
        "geojson": Object {
          "__typename": "Geojson",
          "osm": Object {
            "__typename": "OpenStreetMap",
            "features": Array [
              Object {
                "__typename": "geojsonFeature",
                "geometry": Object {
                  "__typename": "GeoJSONLineString",
                  "coordinates": Array [
                    Array [
                      2.3372032,
                      48.8404254,
                    ],
                    Array [
                      2.3372715,
                      48.8406677,
                    ],
                    Array [
                      2.3373962,
                      48.8410543,
                    ],
                    Array [
                      2.3380024,
                      48.8422009,
                    ],
                    Array [
                      2.3385107,
                      48.8431117,
                    ],
                    Array [
                      2.3389787,
                      48.8439785,
                    ],
                    Array [
                      2.3392577,
                      48.8444515,
                    ],
                    Array [
                      2.3402917,
                      48.8463852,
                    ],
                    Array [
                      2.3405962,
                      48.8469464,
                    ],
                    Array [
                      2.3407859,
                      48.8472594,
                    ],
                    Array [
                      2.3411775,
                      48.8479028,
                    ],
                    Array [
                      2.3422987,
                      48.8497393,
                    ],
                    Array [
                      2.34276,
                      48.8507232,
                    ],
                    Array [
                      2.3429223,
                      48.8510339,
                    ],
                    Array [
                      2.3430309,
                      48.8511954,
                    ],
                    Array [
                      2.3432019,
                      48.8514041,
                    ],
                    Array [
                      2.3434513,
                      48.851675,
                    ],
                    Array [
                      2.3436619,
                      48.851859,
                    ],
                    Array [
                      2.3439871,
                      48.8520933,
                    ],
                    Array [
                      2.3443787,
                      48.8523492,
                    ],
                    Array [
                      2.3447636,
                      48.8525821,
                    ],
                    Array [
                      2.3454415,
                      48.8529788,
                    ],
                    Array [
                      2.3458338,
                      48.8532131,
                    ],
                    Array [
                      2.346153,
                      48.8534072,
                    ],
                    Array [
                      2.3465238,
                      48.8536327,
                    ],
                    Array [
                      2.3468805,
                      48.8538939,
                    ],
                    Array [
                      2.3472815,
                      48.8542327,
                    ],
                    Array [
                      2.3476597,
                      48.8546377,
                    ],
                  ],
                  "type": "LineString",
                  Symbol(id): "$geojsonFeature:way/29735335.geometry",
                },
                "id": "way/29735335",
                "properties": Object {
                  "@changeset": "40980697",
                  "@id": "way/29735335",
                  "@timestamp": "2016-07-23T22:33:32Z",
                  "@uid": "158826",
                  "@user": "cquest",
                  "@version": "34",
                  "electrified": "contact_line",
                  "frequency": "0",
                  "gauge": "1435",
                  "importance": "regional",
                  "layer": "-5",
                  "name": "RER B",
                  "operator": "RATP",
                  "railway": "rail",
                  "source": "STIF",
                  "tunnel": "yes",
                  "usage": "main",
                  "voltage": "1500",
                },
                "type": "Feature",
                Symbol(id): "geojsonFeature:way/29735335",
              },
              Object {
                "__typename": "geojsonFeature",
                "geometry": Object {
                  "__typename": "GeoJSONLineString",
                  "coordinates": Array [
                    Array [
                      2.3580437,
                      48.8489285,
                    ],
                    Array [
                      2.3569883,
                      48.8493417,
                    ],
                    Array [
                      2.3566417,
                      48.8494774,
                    ],
                    Array [
                      2.3498091,
                      48.8517476,
                    ],
                    Array [
                      2.3481873,
                      48.85241,
                    ],
                    Array [
                      2.3468858,
                      48.8530098,
                    ],
                    Array [
                      2.3465228,
                      48.8531476,
                    ],
                    Array [
                      2.3454232,
                      48.8534731,
                    ],
                  ],
                  "type": "LineString",
                  Symbol(id): "$geojsonFeature:way/334870919.geometry",
                },
                "id": "way/334870919",
                "properties": Object {
                  "@changeset": "34996941",
                  "@id": "way/334870919",
                  "@timestamp": "2015-10-31T17:43:46Z",
                  "@uid": "53964",
                  "@user": "_Mathieu_",
                  "@version": "4",
                  "electrified": "contact_line",
                  "frequency": "0",
                  "gauge": "1435",
                  "importance": "regional",
                  "layer": "-1",
                  "maxspeed": "60",
                  "name": "RER C - Ligne de Musée d'Orsay à Austerlitz",
                  "railway": "rail",
                  "tracks": "2",
                  "tunnel": "yes",
                  "usage": "main",
                  "voltage": "1500",
                },
                "type": "Feature",
                Symbol(id): "geojsonFeature:way/334870919",
              },
              Object {
                "__typename": "geojsonFeature",
                "geometry": Object {
                  "__typename": "GeoJSONLineString",
                  "coordinates": Array [
                    Array [
                      2.3454232,
                      48.8534731,
                    ],
                    Array [
                      2.344675,
                      48.8536946,
                    ],
                    Array [
                      2.3440745,
                      48.8539094,
                    ],
                    Array [
                      2.343693,
                      48.8540843,
                    ],
                    Array [
                      2.3431388,
                      48.854372,
                    ],
                    Array [
                      2.342491,
                      48.8547437,
                    ],
                    Array [
                      2.3402755,
                      48.8563525,
                    ],
                    Array [
                      2.3379956,
                      48.8575207,
                    ],
                    Array [
                      2.3329547,
                      48.8584095,
                    ],
                    Array [
                      2.3275778,
                      48.8601289,
                    ],
                  ],
                  "type": "LineString",
                  Symbol(id): "$geojsonFeature:way/377746412.geometry",
                },
                "id": "way/377746412",
                "properties": Object {
                  "@changeset": "34996941",
                  "@id": "way/377746412",
                  "@timestamp": "2015-10-31T17:43:41Z",
                  "@uid": "53964",
                  "@user": "_Mathieu_",
                  "@version": "1",
                  "electrified": "contact_line",
                  "frequency": "0",
                  "gauge": "1435",
                  "importance": "regional",
                  "layer": "-1",
                  "maxspeed": "50",
                  "name": "RER C - Ligne de Musée d'Orsay à Austerlitz",
                  "railway": "rail",
                  "tracks": "2",
                  "tunnel": "yes",
                  "usage": "main",
                  "voltage": "1500",
                },
                "type": "Feature",
                Symbol(id): "geojsonFeature:way/377746412",
              },
              Object {
                "__typename": "geojsonFeature",
                "geometry": Object {
                  "__typename": "GeoJSONPoint",
                  "coordinates": Array [
                    2.3454363,
                    48.8534806,
                  ],
                  "type": "Point",
                  Symbol(id): "$geojsonFeature:node/3572156993.geometry",
                },
                "id": "node/3572156993",
                "properties": Object {
                  "@changeset": "42352341",
                  "@id": "node/3572156993",
                  "@timestamp": "2016-09-22T16:51:49Z",
                  "@uid": "1429581",
                  "@user": "Guillaume Mortier",
                  "@version": "3",
                  "STIF:zone": "1",
                  "name": "Saint-Michel Notre-Dame",
                  "official_name": "ST MICHEL NOTRE DAME",
                  "operator": "SNCF;RATP",
                  "railway": "station",
                  "ref": "RER B, C",
                  "ref:FR:RATP": "5604",
                  "ref:FR:uic8": "87547315;87785436",
                  "ref:SNCF": "RER",
                  "ref:SNCF:RER": "C",
                  "type:RATP": "rer",
                  "uic_ref": "8754731;8778543",
                  "wheelchair": "yes",
                  "wikipedia": "fr:Gare Saint-Michel - Notre-Dame",
                },
                "type": "Feature",
                Symbol(id): "geojsonFeature:node/3572156993",
              },
              Object {
                "__typename": "geojsonFeature",
                "geometry": Object {
                  "__typename": "GeoJSONPoint",
                  "coordinates": Array [
                    2.3402917,
                    48.8463852,
                  ],
                  "type": "Point",
                  Symbol(id): "$geojsonFeature:node/327432589.geometry",
                },
                "id": "node/327432589",
                "properties": Object {
                  "@changeset": "40980697",
                  "@id": "node/327432589",
                  "@timestamp": "2016-07-23T22:33:35Z",
                  "@uid": "158826",
                  "@user": "cquest",
                  "@version": "12",
                  "STIF:zone": "1",
                  "name": "Luxembourg",
                  "operator": "RATP",
                  "railway": "station",
                  "ref": "RER B",
                  "ref:FR:RATP": "1811",
                  "type:RATP": "rer",
                  "uic_ref": "8775861",
                  "wheelchair": "yes",
                  "wikipedia": "fr:Gare du Luxembourg",
                },
                "type": "Feature",
                Symbol(id): "geojsonFeature:node/327432589",
              },
              Object {
                "__typename": "geojsonFeature",
                "geometry": Object {
                  "__typename": "GeoJSONPoint",
                  "coordinates": Array [
                    2.346153,
                    48.8534072,
                  ],
                  "type": "Point",
                  Symbol(id): "$geojsonFeature:node/270191809.geometry",
                },
                "id": "node/270191809",
                "properties": Object {
                  "@changeset": "42352341",
                  "@id": "node/270191809",
                  "@timestamp": "2016-09-22T16:51:50Z",
                  "@uid": "1429581",
                  "@user": "Guillaume Mortier",
                  "@version": "29",
                  "STIF:zone": "1",
                  "name": "Saint-Michel Notre-Dame",
                  "official_name": "ST MICHEL NOTRE DAME",
                  "operator": "SNCF;RATP",
                  "public_transport": "stop_position",
                  "railway": "stop",
                  "ref": "RER B, C",
                  "ref:FR:RATP": "5604",
                  "ref:FR:uic8": "87547315;87785436",
                  "ref:SNCF": "RER",
                  "ref:SNCF:RER": "C",
                  "type:RATP": "rer",
                  "uic_ref": "8754731;8778543",
                  "wheelchair": "yes",
                  "wikipedia": "fr:Gare Saint-Michel - Notre-Dame",
                },
                "type": "Feature",
                Symbol(id): "geojsonFeature:node/270191809",
              },
              Object {
                "__typename": "geojsonFeature",
                "geometry": Object {
                  "__typename": "GeoJSONPoint",
                  "coordinates": Array [
                    2.3569883,
                    48.8493417,
                  ],
                  "type": "Point",
                  Symbol(id): "$geojsonFeature:node/3809893083.geometry",
                },
                "id": "node/3809893083",
                "properties": Object {
                  "@changeset": "34971642",
                  "@id": "node/3809893083",
                  "@timestamp": "2015-10-30T12:46:44Z",
                  "@uid": "53964",
                  "@user": "_Mathieu_",
                  "@version": "1",
                  "highway": "crossing",
                },
                "type": "Feature",
                Symbol(id): "geojsonFeature:node/3809893083",
              },
            ],
            Symbol(id): "$Region:belgium.geojson.osm",
          },
          Symbol(id): "$Region:belgium.geojson",
        },
        "id": "belgium",
        Symbol(id): "Region:belgium",
      },
      Symbol(id): "$ROOT_QUERY.store",
    },
    Symbol(id): "ROOT_QUERY",
  },
  "loading": false,
  "networkStatus": 7,
  "stale": false,
}
`;

exports[`MapboxContainer render 1`] = `
Object {
  "data": Object {
    "error": undefined,
    "fetchMore": [Function],
    "loading": true,
    "mapbox": Object {
      "iconAtlas": "data/location-icon-atlas.png",
      "mapboxApiAccessToken": "pk.eyJ1IjoiY2Fsb2NhbiIsImEiOiJjaXl1aXkxZjkwMG15MndxbmkxMHczNG50In0.07Zu3XXYijL6GJMuxFtvQg",
      "preventStyleDiffing": false,
      "showCluster": "true",
      "showZoomControls": true,
    },
    "networkStatus": 1,
    "refetch": [Function],
    "region": Object {
      "id": "belgium",
      "mapbox": Object {
        "viewport": Object {
          "latitude": 50.5915,
          "longitude": 2.0165,
          "zoom": 5,
        },
      },
      "name": "Belgium",
    },
    "settings": Object {
      "api": Object {
        "host": "localhost",
        "port": "3000",
        "protocol": "http",
        "root": "/graphql/",
      },
      "cycle": Object {
        "drivers": Object {
          "api": "HTTP",
        },
      },
      "domain": "localhost",
      "graphcool": Object {
        "authTokenKey": "graphcool-auth-token",
        "linksPerPage": 5,
        "serviceIdKey": "cjajyycub38710185wt87zsm8",
        "userId": "graphcool-user-id",
      },
      "mapbox": Object {
        "iconAtlas": "data/location-icon-atlas.png",
        "mapboxApiAccessToken": "pk.eyJ1IjoiY2Fsb2NhbiIsImEiOiJjaXl1aXkxZjkwMG15MndxbmkxMHczNG50In0.07Zu3XXYijL6GJMuxFtvQg",
        "preventStyleDiffing": false,
        "showCluster": "true",
        "showZoomControls": true,
      },
      "markers": Object {},
      "overpass": Object {
        "cellSize": 100,
        "sleepBetweenCalls": 1000,
      },
    },
    "startPolling": [Function],
    "stopPolling": [Function],
    "subscribeToMore": [Function],
    "updateQuery": [Function],
    "user": Object {
      "id": "belgiumManager",
      "isActive": true,
      "name": "Belgium Manager",
      "permissions": Array [
        "manage",
      ],
      "regions": Array [
        Object {
          "id": "belgium",
          "isSelected": true,
        },
      ],
    },
    "variables": Object {
      "regionId": "belgium",
    },
    "viewport": Object {
      "bearing": 0,
      "isDragging": false,
      "latitude": 50.5915,
      "longitude": 2.0165,
      "pitch": 40,
      "startDragLngLat": null,
      "zoom": 5,
    },
  },
  "style": Object {
    "MozAppearance": "none",
    "WebkitAppearance": "none",
    "color": "black",
    "fontFamily": "inherit",
    "fontSize": "inherit",
    "height": -32,
    "textDecoration": "none",
    "width": -32,
  },
}
`;

exports[`MapboxContainer render 2`] = `
Object {
  "children": <InteractiveMap
    altitude={1.5}
    attributionControl={true}
    bearing={0}
    clickRadius={0}
    doubleClickZoom={true}
    dragPan={true}
    dragRotate={true}
    getCursor={[Function]}
    height={-32}
    hoverMarker={undefined}
    iconAtlas="data/location-icon-atlas.png"
    isDragging={false}
    latitude={50.5915}
    longitude={2.0165}
    mapStyle="mapbox://styles/mapbox/light-v8"
    mapboxApiAccessToken="pk.eyJ1IjoiY2Fsb2NhbiIsImEiOiJjaXl1aXkxZjkwMG15MndxbmkxMHczNG50In0.07Zu3XXYijL6GJMuxFtvQg"
    maxLatitude={85.05113}
    maxPitch={60}
    maxZoom={20}
    minLatitude={-85.05113}
    minPitch={0}
    minZoom={0}
    onClick={null}
    onError={[Function]}
    onHover={null}
    onLoad={[Function]}
    onViewportChange={null}
    pitch={40}
    preserveDrawingBuffer={false}
    preventStyleDiffing={false}
    reuseMaps={false}
    scrollZoom={true}
    selectMarker={undefined}
    showCluster="true"
    showZoomControls={true}
    startDragLngLat={null}
    style={
      Object {
        "height": -32,
        "width": -32,
      }
    }
    touchZoomRotate={true}
    visibilityConstraints={
      Object {
        "maxLatitude": 85.05113,
        "maxPitch": 60,
        "maxZoom": 20,
        "minLatitude": -85.05113,
        "minPitch": 0,
        "minZoom": 0,
      }
    }
    visible={true}
    width={-32}
    zoom={5}
  />,
  "className": "mapbox-map-react-gl-outer",
}
`;

exports[`MapboxContainer renderError 1`] = `
Object {
  "data": Object {
    "error": undefined,
    "fetchMore": [Function],
    "loading": true,
    "mapbox": Object {
      "iconAtlas": "data/location-icon-atlas.png",
      "mapboxApiAccessToken": "pk.eyJ1IjoiY2Fsb2NhbiIsImEiOiJjaXl1aXkxZjkwMG15MndxbmkxMHczNG50In0.07Zu3XXYijL6GJMuxFtvQg",
      "perspectiveEnabled": true,
      "preventStyleDiffing": false,
      "showCluster": "true",
      "showZoomControls": true,
    },
    "networkStatus": 1,
    "refetch": [Function],
    "region": Object {
      "id": "foo",
      "mapbox": Object {
        "viewport": Object {
          "latitude": 37,
          "longitude": -119,
          "zoom": 5,
        },
      },
      "name": "Oakland",
    },
    "settings": Object {
      "api": Object {
        "host": "localhost",
        "port": "3000",
        "protocol": "http",
        "root": "/graphql/",
      },
      "cycle": Object {
        "drivers": Object {
          "api": "HTTP",
        },
      },
      "domain": "localhost",
      "graphcool": Object {
        "authTokenKey": "graphcool-auth-token",
        "linksPerPage": 5,
        "serviceIdKey": "cjajyycub38710185wt87zsm8",
        "userId": "graphcool-user-id",
      },
      "mapbox": Object {
        "iconAtlas": "data/location-icon-atlas.png",
        "mapboxApiAccessToken": "pk.eyJ1IjoiY2Fsb2NhbiIsImEiOiJjaXl1aXkxZjkwMG15MndxbmkxMHczNG50In0.07Zu3XXYijL6GJMuxFtvQg",
        "perspectiveEnabled": true,
        "preventStyleDiffing": false,
        "showCluster": "true",
        "showZoomControls": true,
      },
      "markers": Object {},
      "overpass": Object {
        "cellSize": 100,
        "sleepBetweenCalls": 1000,
      },
    },
    "startPolling": [Function],
    "stopPolling": [Function],
    "subscribeToMore": [Function],
    "updateQuery": [Function],
    "user": Object {
      "id": "californiaManager",
      "isActive": true,
      "name": "Jerry Brown",
      "permissions": Array [
        "manage",
      ],
      "regions": Array [
        Object {
          "id": "oakland",
          "isSelected": true,
        },
      ],
    },
    "variables": Object {
      "regionId": "foo",
    },
    "viewport": Object {
      "latitude": 37,
      "longitude": -119,
      "zoom": 5,
    },
  },
  "style": Object {
    "MozAppearance": "none",
    "WebkitAppearance": "none",
    "backgroundColor": "blue",
    "border": 0,
    "color": "white",
    "display": "inline-block",
    "fontFamily": "inherit",
    "fontSize": "inherit",
    "fontWeight": "bold",
    "height": 720,
    "margin": 0,
    "paddingBottom": 8,
    "paddingLeft": 16,
    "paddingRight": 16,
    "paddingTop": 8,
    "textDecoration": "none",
    "width": 1080,
  },
}
`;
